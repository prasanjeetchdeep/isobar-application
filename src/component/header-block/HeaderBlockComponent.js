import React, {Fragment} from 'react';
import { Navbar, Button, Nav, Container, Jumbotron, Badge } from "react-bootstrap";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShoppingCart } from '@fortawesome/free-solid-svg-icons'
import SearchBoxComponent from "../search-box/SearchBoxComponent";

const HeaderBlockComponent = ({
    onSearch
}) => {
return(
    <Fragment>
        <Navbar className="navigation-bar" bg="light" expand="lg">
        <Navbar.Brand href="#home">
          <img
            class="header-logo-image"
            src="https://appexchange.salesforce.com/servlet/servlet.FileDownload?file=00P3A00000a2QcXUAU"
            alt="logo"
          />
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto"></Nav>
          <span classname="header-options">
            <span style={{ position: "relative" }}>
              <FontAwesomeIcon icon={faShoppingCart} /> <Badge variant="danger" className="badge-count" >3</Badge>
            </span>
            <Button variant="outline-primary" style={{ marginLeft: 15 }}>Sign in</Button>

          </span>

        </Navbar.Collapse>
      </Navbar>
      <Container><Jumbotron className="jumbotron-background">
        <div style={{ maxWidth: 500 }}>
          <h1 style={{ color: "#e8e6e6" }}>A special thank you</h1>
          <p style={{ color: "#c7c7c7" }}>We appreciate students like you. Get courses from â‚¹432! Ends Dec. 26.</p>

          <SearchBoxComponent
            className="header-search-box"
            onSearch={search => onSearch(search)}
          />
        </div>
      </Jumbotron>
      </Container>
        </Fragment>
);
};

export default HeaderBlockComponent;