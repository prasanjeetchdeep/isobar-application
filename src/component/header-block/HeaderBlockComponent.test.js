import React from "react";
import { render } from "@testing-library/react";
import HeaderBlockComponent from "./HeaderBlockComponent";

const noop = () => {};
const cart = [1];
const books = [
    {
      id: 1,
      name: "React - basics",
      description: "This course is going to take you through basics of React.",
      author: "James White",
      publishDate: "12/03/2019",
      duration: "00:03:56",
      image: "https://cdn.auth0.com/blog/react-js/react.png"
    }
  ];
  

it("should render HeaderBlockComoponent with 1 item in cart", () => {
  const { asFragment } = render(
    <HeaderBlockComponent
      onSignInClick={noop}
      onSignOutClick={noop}
      onSearch={noop}
      signedIn={true}
      cart={cart}
      books={books}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should render HeaderBlockComoponent with 1 item in cart without being signedin", () => {
    const { asFragment } = render(
      <HeaderBlockComponent
        onSignInClick={noop}
        onSignOutClick={noop}
        onSearch={noop}
        signedIn={false}
        cart={cart}
        books={books}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
