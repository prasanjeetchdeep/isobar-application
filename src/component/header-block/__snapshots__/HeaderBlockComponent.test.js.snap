// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeaderBlockComoponent with 1 item in cart 1`] = `
<DocumentFragment>
  <nav
    class="navigation-bar navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      class="navbar-brand"
      href="#home"
    >
      <img
        alt="logo"
        class="header-logo-image"
        src="https://appexchange.salesforce.com/servlet/servlet.FileDownload?file=00P3A00000a2QcXUAU"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      />
      <span
        class="header-options"
      >
        <span
          class="relative-position"
        >
          <span
            class="cursor-pointer"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-shopping-cart fa-w-18 "
              data-icon="shopping-cart"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                fill="currentColor"
              />
            </svg>
             
            <span
              class="badge-count badge badge-danger"
            >
              1
            </span>
          </span>
        </span>
        <button
          class="btn btn-outline-danger"
          style="margin-left: 15px;"
          type="button"
        >
          Sign out
        </button>
      </span>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="jumbotron-background jumbotron"
    >
      <div
        class="search-jt-container"
        style="max-width: 500px;"
      >
        <h1
          class="jumbotron-title"
        >
          A special thank you
        </h1>
        <p
          class="jumbotron-subtitle"
        >
          We appreciate students like you. Get courses now! Ends Dec. 31
        </p>
        <div
          class="flexbox search-container"
        >
          <img
            alt="Search"
            class="search-icon"
            src="Search.svg"
          />
          <input
            class="search-box input-font"
            placeholder="What do you want to learn?"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render HeaderBlockComoponent with 1 item in cart without being signedin 1`] = `
<DocumentFragment>
  <nav
    class="navigation-bar navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      class="navbar-brand"
      href="#home"
    >
      <img
        alt="logo"
        class="header-logo-image"
        src="https://appexchange.salesforce.com/servlet/servlet.FileDownload?file=00P3A00000a2QcXUAU"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      />
      <span
        class="header-options"
      >
        <span
          class="relative-position"
        >
          <span
            class="cursor-pointer"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-shopping-cart fa-w-18 "
              data-icon="shopping-cart"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                fill="currentColor"
              />
            </svg>
             
            <span
              class="badge-count badge badge-danger"
            >
              1
            </span>
          </span>
        </span>
        <button
          class="btn btn-outline-primary"
          style="margin-left: 15px;"
          type="button"
        >
          Sign in
        </button>
      </span>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="jumbotron-background jumbotron"
    >
      <div
        class="search-jt-container"
        style="max-width: 500px;"
      >
        <h1
          class="jumbotron-title"
        >
          A special thank you
        </h1>
        <p
          class="jumbotron-subtitle"
        >
          We appreciate students like you. Get courses now! Ends Dec. 31
        </p>
        <div
          class="flexbox search-container"
        >
          <img
            alt="Search"
            class="search-icon"
            src="Search.svg"
          />
          <input
            class="search-box input-font"
            placeholder="What do you want to learn?"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
